tests/validation/tests/v3_api/test_bkp_restore_s3_recover.py

```diff
@@ -0,0 +1,156 @@
+    assert len(etcd_nodes) > 0, "Make sure we have etcd nodes in the cluster"
+
+    # Proceed to remove all etcd nodes
+    for etcd_node in etcd_nodes:
+        ips_to_remove.append(etcd_node.externalIpAddress)
+        client.delete(etcd_node)
+    # Also remove the ec2 instances
+    for ip_to_remove in ips_to_remove:
+        delete_node_from_ec2(ip_to_remove)
+        namespace["nodes"] = [node for node
+                              in namespace["nodes"]
+                              if node.public_ip_address != ip_to_remove]
+    ips_to_remove.clear()
+    cluster = client.reload(cluster)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) == 0, "Ensure the cluster has zero etcd nodes now"
+    # Add completely new etcd nodes to the cluster
+    cluster = add_new_etcd_nodes(client, cluster)
+    cluster = client.reload(cluster)
+    # This message is expected to appear after we add new etcd nodes
+    # The cluster will require the user to perform a backup to recover
+    # this is appears in the cluster object in cluster.transitioningMessage
+    message = "Please restore your cluster from backup"
+    cluster = wait_for_cluster_transitioning_message(client, cluster, message)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) == 3, "Make sure the cluster now has 3 etcd nodes"
+    cluster = restore_backup_to_recover(client, cluster, b_info)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    worker_node = get_worker_nodes(client, cluster)
+    # validate the ingress that was created in the first backup
+    # after restoring and recovering the cluster
+    validate_ingress(namespace["p_client"], cluster,
+                     [b_info["workload"]], ns["host"],
+                     "/name.html")
+
+
+@pytest.fixture(scope='module', autouse=True)
+def create_project_client_and_cluster_s3_three_etcd(request):
+    node_roles = [
+        ["controlplane", "worker"],
+        ["etcd"], ["etcd"], ["etcd"],
+    ]
+    cluster, aws_nodes = create_and_validate_custom_host(node_roles)
+    client = get_user_client()
+    namespace["nodes"].extend(aws_nodes)
+
+    p, ns = create_project_and_ns(USER_TOKEN, cluster, "testrecover")
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    c_client = get_cluster_client_for_token(cluster, USER_TOKEN)
+
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+    namespace["c_client"] = c_client
+    namespace["client"] = client
+
+    def fin():
+        cluster_cleanup(client, cluster, namespace["nodes"])
+    request.addfinalizer(fin)
+
+
+def wait_for_cluster_transitioning_message(client, cluster, message):
+    start = time.time()
+    while message not in cluster.transitioningMessage:
+        print(cluster.transitioningMessage)
+        time.sleep(5)
+        cluster = client.reload(cluster)
+        if time.time() - start > DEFAULT_TIMEOUT:
```

Comment:
Is DEFAULT_TIMEOUT(2 mins) always sufficient? In my testing I have seen it take longer than that.